generation/maisi/scripts/compute_fid_2-5d_ct.py

```diff
@@ -0,0 +1,655 @@
+
+        images_2d = torch.cat(slices, dim=0)
+        images_2d = radimagenet_intensity_normalisation(images_2d)
+        images_2d = images_2d[mapping_index]
+
+        feature_image_yz = feature_network.forward(images_2d)
+        feature_image_yz = spatial_average(feature_image_yz, keepdim=False)
+
+        # ---------------------- ZX-plane slicing along W ----------------------
+        if center_slices:
+            start_w = int((1.0 - center_slices_ratio) / 2.0 * W)
+            end_w = int((1.0 + center_slices_ratio) / 2.0 * W)
+            slices = torch.unbind(image[:, :, :, start_w:end_w:sample_every_k, :], dim=3)
+        else:
+            slices = torch.unbind(image, dim=3)
+
+        if drop_empty:
+            mapping_index = drop_empty_slice(slices, empty_threshold)
+        else:
+            mapping_index = [True for _ in range(len(slices))]
+
+        images_2d = torch.cat(slices, dim=0)
+        images_2d = radimagenet_intensity_normalisation(images_2d)
+        images_2d = images_2d[mapping_index]
+
+        feature_image_zx = feature_network.forward(images_2d)
+        feature_image_zx = spatial_average(feature_image_zx, keepdim=False)
+
+    return feature_image_xy, feature_image_yz, feature_image_zx
+
+
+def pad_to_max_size(tensor: torch.Tensor, max_size: int, padding_value: float = 0.0) -> torch.Tensor:
+    """
+    Zero-pad a 2D feature map or other tensor along the first dimension to match a specified size.
+
+    Args:
+        tensor (torch.Tensor): The feature tensor to pad.
+        max_size (int): Desired size along the first dimension.
+        padding_value (float): Value to fill during padding.
+
+    Returns:
+        torch.Tensor: Padded tensor matching `max_size` along dim=0.
+    """
+    pad_size = [0, 0] * (len(tensor.shape) - 1) + [0, max_size - tensor.shape[0]]
+    return F.pad(tensor, pad_size, "constant", padding_value)
+
+
+def main(
+    data0_dataroot: str = "path/to/datasetA",
+    data0_filelist: str = "path/to/filelistA.txt",
+    data0_folder: str = "datasetA",
+    data1_dataroot: str = "path/to/datasetB",
+    data1_filelist: str = "path/to/filelistB.txt",
+    data1_folder: str = "datasetB",
+    enable_center_slices: bool = False,
+    enable_center_slices_ratio: float = 0.4,
+    enable_padding: bool = True,
+    enable_center_cropping: bool = True,
+    enable_resampling: bool = False,
+    enable_resampling_spacing: str = "1.0x1.0x1.0",
+    ignore_existing: bool = False,
+    model_name: str = "radimagenet_resnet50",
+    num_images: int = 100,
+    output_root: str = "./features/features-512x512x512",
+    target_shape: str = "512x512x512",
+):
+    """
+    Main function to compute 2.5D features for two datasets (e.g. real vs. synthetic)
+    and calculate FID across XY, YZ, ZX planes.
+
+    Args:
+        data0_dataroot (str): Root path of dataset 0 (real images).
+        data0_filelist (str): Text file listing the 3D images in dataset 0.
```

Comment:
Can we give an small example for what should this filelist txt looks like in the docstring?

Reply:
Thanks for suggestion. Changes have been made.